src/rpdk/core/contract/suite/handler_commons.py

```diff
@@ -124,3 +125,13 @@ def test_delete_failure_not_found(resource_client, current_resource_model):
         Action.DELETE, OperationStatus.FAILED, current_resource_model
     )
     return error_code
+
+
+def test_input_equals_output(resource_client, input_model, output_model):
+    prune_properties_from_model(input_model, resource_client.create_only_paths)
+    prune_properties_from_model(input_model, resource_client.write_only_paths)
+
+    prune_properties_from_model(output_model, resource_client.read_only_paths)
+    prune_properties_from_model(output_model, resource_client.create_only_paths)
```

Comment:
Why would [`createOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) be pruned from both input and output models?

Reply:
because, create only property can be entered by the customer or generated by the handler. The test will intermittently depending on how this was enetered

Reply:
> or generated by the handler

This should be covered by pruning [`readOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) already. [`createOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) really shouldn't be [`readOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) anyway, but I suspect some [`createOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) are being incorrectly marked as [`readOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) as well for [`GetAtt` support](https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/68#issuecomment-665243220)

Reply:
@anshikg are those conditionally [`readOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) limited to [`primaryIdentifier`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics)? Would filtering [`primaryIdentifier`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) instead of [`createOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) make sense?

Reply:
That is an interesting point. I have only seen createOnlyProperties which are primaryIdentifiers being generated. But I am not sure if this is a norm. @jotompki @rjlohan is that the case?

Reply:
> This should be covered by pruning readOnlyProperties already. createOnlyProperties really shouldn't be readOnlyProperties anyway, but I suspect some createOnlyProperties are being incorrectly marked as readOnlyProperties as well for GetAtt support

there might be some primary identifiers that are not required and in this case create handler will have a method to generate them (as to my knowledge we do not generate any other types of create onlies). However, we should treat such primary id as provided by the customer (as we do it for them)

Reply:
sure. But, we wont have those ids in the input. Which will lead to input != output

Reply:
No they should be just treated as normal properties and not as read only properties.

Reply:
As dicussed offline:
> I am not treating createOnly property as readOnly. The reason I pruned createOnly property is because, createOnly  property can be entered by the customer or by the handlers depending on how it was entered the contract test will pass or fail. Therefore, I took it out.

Reply:
*If* `createOnlyProperties` are specified in the input, they should be equal on the output. Otherwise, we can ignore them as you've done.

Reply:
makes sense. I will add the check

Reply:
Now that I think about it more, conditionally [`readOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics) can apply to any type of property that has a default value, and we might have to ignore any property returned that was not in the input

(example: [`AWS::DynamoDB::Table.BillingMode`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-billingmode))